### DISCUSSION TITLE
Evolving Demographics with Evolving Population Features

### DISCUSSION THREAD

**Discussion by AmmarFaiq** [synthetichealth/synthea#1045] [General]
### Requested Feature

Hello everyone,

I was looking at the Evolving population features and trying to understand a little bit how it works. Especially regarding the "clock" system and the "snapshot" input. 
Based on some comments on issue #716 it seems that it is possible to change demographics yearly with the "Evolving population" features by changing the exporter years of history into 1 year in the synthetic properties. My questions are :

1. Does "Evolving population" works with CSV output? and do we need to change the CSV file or the snap file?
2. Does "Evolving population" works with the INCOME, SOCIO_ECONOMIC_CATEGORY, EDUCATION, and "MARITAL STATUS"?
3. Does a change in CSV demographics data in each snapshot folder going to change the model behaviour? (assuming that I already change some modules to add demographic changes to patient treatment). 
eg. Assuming patient 1 in snapshot 1 belongs to Low SOCIOECONOMIC CATEGORY, then I change it on snapshot 2 to High. Does the third snapshot output treatment behaviour going to change?
4. Supposed I want to create data from 2013 - 2022 with changing demographic every year. How do we define that the first snapshot is a model that only simulates patients for the year 2013?

Thanks,

Ammar

**Comment by hadleynet** [comment]
Hi Ammar,

1. Evolving population should work with any of the export formats. The snapshot file captures a binary representation of the state of the simulation at the end of a run and can subsequently be reloaded to "continue" the run into the future.
2, 3, 4. Demographics information is assigned to each simulated patient at "birth", changing the CSV demographics data between runs while using the evolving population feature won't change the demographics assigned to a given patient.

Hope this helps to clarify how the evolving population feature works right now.

Marc.

**Reply by AmmarFaiq** [reply]
Thanks for the quick response!
As for question number 4. Do we have some kind of features or code, that can set the specific year of the simulation (eg. start the simulation in 2013)?

**Reply by jawalonoski** [reply]
@eedrummer can you comment on @AmmarFaiq question number 4 with respect to [Fixed Records](https://github.com/synthetichealth/synthea/wiki/Fixed-Records)? Is it possible that they could swap out different fixed records for each year, generate them separately, and combine them? I'm thinking that would work for demographic data, but not clinical data? Any other issues with that?

@AmmarFaiq another option to change socio economic status is to just write a module that does that... and as each year goes by, it uses whatever logic you include in the module to override those patient attributes... if/when they are next used, the simulation will make different choices... For example, a wealthy person becomes poor, and no longer can afford an insurance plan.

**Reply by AmmarFaiq** [reply]
@jawalonoski That's a great idea for the module. Suppose I want to follow that solution, any idea where I can put the change of socioeconomic records ?

**Reply by jawalonoski** [reply]
Meaning you want to record the socioeconomic status? Use an Observation. Not sure you'll find a matching LOINC code, but if you can't, just make one up for your data analysis.

**Reply by jawalonoski** [reply]
If you meant, how do you override the socioeconomic status? Use a `SetAttribute` state, and specify each attribute (the string name has to match exactly) and value.

**Reply by AmmarFaiq** [reply]
Got it!. Thanks a lot everyone!

**Comment by PAastha** [comment]
Hi, I am trying to generate more information for same patient, and while giving -t flag as 1 since I want data simulated daily for the patient, i am getting the endtime before referencetime in metadata,could this be a problem for simulation ofpatient information?
